iosapp/lib-ios/Sources/LibClients/DeviceClient.swift

```diff
@@ -7,18 +7,20 @@ import Foundation
 #endif
 
 public struct DeviceClient: Sendable {
-  public var type: DeviceType
-  public var iOSVersion: String
-  public var vendorId: UUID?
-  public var batteryLevel: @Sendable () async -> BatteryLevel
+  @MainActor public var type: @Sendable () async -> DeviceType
+  @MainActor public var iOSVersion: @Sendable () async -> String
+  @MainActor public var vendorId: @Sendable () async -> UUID?
+  @MainActor public var data: @Sendable () async -> Data
+  @MainActor public var batteryLevel: @Sendable () async -> BatteryLevel
```

Comment:
All these things reach out to `UIDevice.current` which is main actor only, so making these fields `@MainActor` is sorta necessary. i explored some workarounds for this, but they seemed complicated. like for instance, this repo:

https://github.com/tgrapperon/swift-dependencies-additions/blob/main/Sources/DeviceDependency/UIDeviceDependency_iOS.swift

has some crazy main-actor-proxy types to make working with these sorts of deps more ergonomic, but i didn't feel like pulling that in, or going quite that far.